core/src/main/java/com/google/instrumentation/stats/StatsContextFactory.java

```diff
@@ -35,4 +40,33 @@
    * Returns the default {@link StatsContext}.
    */
   public abstract StatsContext getDefault();
+
+  /**
+   * Get current StatsContext from current gRPC Context.
+   *
+   * @return the current {@code StatsContext} from {@code io.grpc.Context}, or {@code null}
+   *     if there's no {@code StatsContext} associated with current {@code io.grpc.Context}.
+   */
+  @Nullable
+  public final StatsContext getCurrentStatsContext() {
+    // TODO(songya): return a no-op StatsContext (or default) when getCurrentStatsContext() returns
+    // null?
+    return ContextUtils.getCurrentStatsContext();
+  }
+
+  /**
+   * Enters the scope of code where the given {@link StatsContext} is in the current Context, and
+   * returns an object that represents that scope. The scope is exited when the returned object is
+   * closed.
+   *
+   * <p>Supports try-with-resource idiom.
+   *
+   * @param statsContext The {@link StatsContext} to be set to the current Context.
+   * @return an object that defines a scope where the given {@link StatsContext} will be set to the
+   *     current Context.
+   * @throws NullPointerException if statsContext is null.
+   */
+  public final NonThrowingCloseable withStatsContext(StatsContext statsContext) {
+    return ContextUtils.withStatsContext(checkNotNull(statsContext, "StatsContext"));
```

Comment:
For consistency with trace we should have the error message exactly the same as the variable name. In this case "statsContext"